src/inc/volatile.h

```diff
@@ -106,26 +106,43 @@
 #define VOLATILE_MEMORY_BARRIER()
 #endif // __GNUC__
 
+template<typename T>
+struct RemoveVolatile
+{
+   typedef T type;
+};
+
+template<typename T>
+struct RemoveVolatile<volatile T>
+{
+   typedef T type;
+};
+
+
 //
 // VolatileLoad loads a T from a pointer to T.  It is guaranteed that this load will not be optimized
 // away by the compiler, and that any operation that occurs after this load, in program order, will
 // not be moved before this load.  In general it is not guaranteed that the load will be atomic, though
 // this is the case for most aligned scalar data types.  If you need atomic loads or stores, you need
 // to consult the compiler and CPU manuals to find which circumstances allow atomicity.
 //
+// Starting at version 3.8, clang errors out on initializing of type int * to volatile int *. To fix this, we add two templates to cast away volatility
```

Comment:
I agree. I actually wanted to move everything to 3.9 in June, but it was not a good time for that as we were getting to zero bugs for 2.0 and everyone was preoccupied with that.
The non-pleasant part of the change is the need to update all of our CI machines to have clang 3.9.